packages/typestack/src/validate-body.hook.spec.ts

```diff
@@ -2,172 +2,186 @@
 import { deepStrictEqual, strictEqual } from 'assert';
 
 // 3p
-import { Context, getHookFunction, isHttpResponseBadRequest, ServiceManager } from '@foal/core';
+import { Class, Context, getHookFunction, HookFunction, isHttpResponseBadRequest, ServiceManager } from '@foal/core';
 import { Expose } from 'class-transformer';
 import { IsNumber, Min } from 'class-validator';
 
 // FoalTS
-import { ValidateBody } from './validate-body.hook';
+import { ValidateBody, ValidateBodyOptions } from './validate-body.hook';
 
 describe('ValidateBody', () => {
 
   class Product {}
-
-  it('should return an HttpResponseBadRequest instance if ctx.request.body is not an object.', async () => {
-    const hook = getHookFunction(ValidateBody(Product));
-    const ctx = new Context({});
-    ctx.request.body = 3;
-
-    const actual = await hook(ctx, new ServiceManager());
-    if (!isHttpResponseBadRequest(actual)) {
-      throw new Error('The hook should have returned an HttpResponseBadRequest instance.');
-    }
-    deepStrictEqual(actual.body, {
-      message: 'The request body should be a valid JSON object or array.'
-    });
-
-    const ctx2 = new Context({});
-    ctx2.request.body = null;
-
-    const actual2 = await hook(ctx2, new ServiceManager());
-    if (!isHttpResponseBadRequest(actual2)) {
-      throw new Error('The hook should have returned an HttpResponseBadRequest instance.');
-    }
-    deepStrictEqual(actual2.body, {
-      message: 'The request body should be a valid JSON object or array.'
-    });
+  context('given class as argument', () => {
```

Comment:
Actually, I didn't change the tests, just generalized them a little bit to run same tests in separate context (to avoid duplication)